plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -432,7 +433,7 @@ private Document getTableMetadata(SchemaTableName schemaTableName)
 
         if (doc == null) {
             if (!collectionExists(db, tableName)) {
-                throw new TableNotFoundException(schemaTableName);
+                throw new TableNotFoundException(new SchemaTableName(schemaName, tableName));
```

Comment:
This will lowercase `schemaName, tableName`. Let's pass actual names (case-preserving) with some `String message` message